### PR TITLE
Updated the cluster definition of the adapter in the native envoy template config

### LINKED ISSUE #106: Adapter address in the envoy config template
Change to `LOGICAL_DNS` and default the address to `localhost`. This will allow more flexibility in cases where both the Envoy and Adapter run in Docker containers.

### PR DESCRIPTION
**Pull request by rockspore**
The type is changed from `static` to `LOGICAL_DNS`. And the host is changed from `127.0.0.1` to `localhost`.

close #106

### CODE REVIEW

**Comment by theganyo** [comment]
This isn't going to be clear. As written, the parameter only affects the native config... but a user could reasonably expect that it would change the host name for the adapter Service name and the EnvoyFilter sidecar reference. Either we need to explicitly call out the inconsistency and give error for the incorrect templates or we need to ensure the flag is consistently applied for all templates.

**Comment by rockspore** [comment]
> This isn't going to be clear. As written, the parameter only affects the native config... but a user could reasonably expect that it would change the host name for the adapter Service name and the EnvoyFilter sidecar reference. Either we need to explicitly call out the inconsistency and give error for the incorrect templates or we need to ensure the flag is consistently applied for all templates.

Good point. `--host` (target service host) and `--tls` fields apply only to native template as well. Meanwhile, `--tag` (adapter image tag) only applies to istio templates. I'd prefer to issue warnings when these flags get specified for wrong templates.

**Comment by theganyo** [comment]
In the `provision` command, we [indicate in the flag description](https://github.com/apigee/apigee-remote-service-cli/blob/master/cmd/provision/provision.go#L115) what they apply to. And I believe we error when they aren't appropriate.

**Comment by rockspore** [comment]
> In the `provision` command, we [indicate in the flag description](https://github.com/apigee/apigee-remote-service-cli/blob/master/cmd/provision/provision.go#L115) what they apply to. And I believe we error when they aren't appropriate.

That's true. I will then add a validation on the flag values.

**Comment by rockspore** [comment]
Added a `samples templates` command to list all supported templates.

Close #108 as well.

File: cmd/samples/samples.go [hunk 1]
```diff
@@ -31,12 +32,28 @@ import (
 	"github.com/spf13/cobra"
 )
 
+var (
+	supportedTemplates = map[string]struct{}{
```
Comment (theganyo) [inline cmd/samples/samples.go:36]: This map might be a good way to map the the template name to the template directory so you can simply look it up from here.

File: cmd/samples/samples.go [hunk 2]
```diff
@@ -100,40 +146,92 @@ files related to deployment of their target services.`,
...
+	c.Flags().StringVarP(&s.TLS.Dir, "tls", "", "", "directory containing tls.key and tls.crt used for the adapter service (envoy templates only)")
+	c.Flags().StringVarP(&s.ImageTag, "tag", "", "", "version tag of the Envoy Adapter image (istio templates only)")
 
 	_ = c.MarkFlagRequired("config")
 
 	return c
 }
 
+func (s *samples) validateFieldsFromFlags() error {
+	if _, ok := supportedTemplates[s.template]; !ok {
```
Comment (theganyo) [inline cmd/samples/samples.go:172]: See comment on using map above so this function can be simplified.

Comment (rockspore) [reply cmd/samples/samples.go:172]: Good idea!

**Review by theganyo — APPROVED** [review]

**Pull request merged**